extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -429,11 +436,49 @@ export class BookTocManager implements IBookTocManager {
 		}
 	}
 
+	public async addNewNotebook(notebookName: string, notebookPath: string, fileType: FileType, book?: BookModel): Promise<void> {
+		if (fileType === FileType.Notebook) {
+			await this.createNotebook(notebookPath.concat(fileType), undefined);
+		} else {
+			await fs.writeFile(notebookPath.concat(fileType), '');
+		}
+		if (book) {
+			let toc = yaml.safeLoad((await fs.readFile(book.tableOfContentsPath, 'utf8')));
+			let notebookToc: JupyterBookSection = {
+				title: notebookName,
+				file: path.posix.join(path.posix.sep, notebookName)
+			};
+			if (book.version === BookVersion.v1) {
+				notebookToc = convertTo(book.version, notebookToc);
+			}
+			toc.push(notebookToc);
+			await fs.writeFile(book.tableOfContentsPath, yaml.safeDump(toc, { lineWidth: Infinity, noRefs: true, skipInvalid: true }));
+		} else {
+			vscode.commands.executeCommand('bookTreeView.openBook', notebookPath.concat(fileType), false, notebookPath.concat(fileType), true);
+		}
+
+	}
+
 	public async removeNotebook(element: BookTreeItem): Promise<void> {
 		const findSection = { file: element.book.page.file, title: element.book.page.title };
 		await this.updateTOC(element.book.version, element.tableOfContentsPath, findSection, undefined);
 	}
 
+
+	public async createNotebook(notebookPath: string, connectionProfile?: azdata.IConnectionProfile): Promise<azdata.nb.NotebookEditor> {
+		await fs.writeFile(notebookPath, '');
+		const notebookUri = vscode.Uri.parse(notebookPath);
+		const options: azdata.nb.NotebookShowOptions = connectionProfile ? {
+			viewColumn: null,
+			preserveFocus: true,
+			preview: null,
```

Comment:
undefined - not null

What's the behavior for all these undefined values anyways? Given you pass it a connection profile shouldn't this be using the SQL kernel?